src/infra/ssl/sslCmdRepo.go

```diff
@@ -79,58 +78,66 @@ func (repo *SslCmdRepo) ReplaceWithSelfSigned(vhostName valueObject.Fqdn) error
-	if err != nil || rawVhostIps == "" {
-		rawVhostIps, err = infraHelper.RunCmd("dig", "+short", vhostStr, "@185.228.168.9")
-		if err != nil || rawVhostIps == "" {
-			return false
-		}
-	}
-
-	rawVhostIpsParts := strings.Split(rawVhostIps, "\n")
-	if len(rawVhostIpsParts) == 0 {
-		return false
-	}
+) (mappingId valueObject.MappingId, err error) {
+	path, _ := valueObject.NewMappingPath(
+		infraData.GlobalConfigs.DomainOwnershipValidationUrlPath,
+	)
+	matchPattern, _ := valueObject.NewMappingMatchPattern("equals")
+	targetType, _ := valueObject.NewMappingTargetType("inline-html")
+	httpResponseCode, _ := valueObject.NewHttpResponseCode(200)
+	targetValue, _ := valueObject.NewMappingTargetValue(
+		expectedOwnershipHash.String(), targetType,
+	)
 
-	var serverIpAddress *valueObject.IpAddress
-	for _, rawVhostIp := range rawVhostIpsParts {
-		ipAddress, err := valueObject.NewIpAddress(rawVhostIp)
-		if err != nil {
-			continue
-		}
+	inlineHtmlMapping := dto.NewCreateMapping(
+		targetVhostName,
+		path,
+		matchPattern,
+		targetType,
+		&targetValue,
+		&httpResponseCode,
+	)
 
-		serverIpAddress = &ipAddress
-		break
+	mappingCmdRepo := mappingInfra.NewMappingCmdRepo(repo.persistentDbSvc)
+	mappingId, err = mappingCmdRepo.Create(inlineHtmlMapping)
+	if err != nil {
+		return mappingId, errors.New(
+			"CreateOwnershipValidationMappingError (" +
+				targetVhostName.String() + "): " + err.Error(),
+		)
 	}
 
-	if serverIpAddress == nil {
-		return false
-	}
+	return mappingId, nil
+}
 
-	hashUrlPath := infraData.GlobalConfigs.DomainOwnershipValidationUrlPath
-	hashUrlFull := "https://" + vhostStr + hashUrlPath
-	serverIpStr := serverIpAddress.String()
+func (repo *SslCmdRepo) filterDomainsMappedToSomewhere(
+	vhostNames []valueObject.Fqdn,
+) []valueObject.Fqdn {
+	domainsMappedToSomewhere := []valueObject.Fqdn{}
+	for _, vhostName := range vhostNames {
+		vhostNameStr := vhostName.String()
 
-	ownershipHashFound, err := infraHelper.RunCmd(
-		"curl", "-skLm", "10", "--resolve", vhostStr+":443:"+serverIpStr, hashUrlFull,
-	)
-	if err != nil {
-		hashUrlFull = "https://" + serverIpStr + hashUrlPath
-		ownershipHashFound, err = infraHelper.RunCmd(
-			"curl", "-skLm", "10", "-H", "Host: "+vhostStr, hashUrlFull,
+		googleDnsServerIp := "8.8.8.8"
+		rawVhostIps, err := infraHelper.RunCmdWithSubShell(
+			"dig +short " + vhostNameStr + " @" + googleDnsServerIp,
 		)
-		if err != nil {
-			return false
+		if err != nil || rawVhostIps == "" {
+			cleanbrowsingSecurityFilterDnsServerIp := "185.228.168.9"
+			rawVhostIps, err := infraHelper.RunCmdWithSubShell(
+				"dig +short " + vhostNameStr + " @" + cleanbrowsingSecurityFilterDnsServerIp,
+			)
+			if err != nil || rawVhostIps == "" {
+				continue
+			}
 		}
+
+		domainsMappedToSomewhere = append(domainsMappedToSomewhere, vhostName)
 	}
 
-	return ownershipHashFound == expectedOwnershipHash.String()
+	return domainsMappedToSomewhere
```

Comment:
Ok, I get, your method name should be filterOnlineDomains or filterFunctioningDomains. That's what you do on the method. MappedToSomewhere gives the idea that you're looking if it's mapped to some place in specific.